Errors when running run_tests.sh

**Describe the issue**
The `run_tests.sh` fails with this error:
```

    def test_document_person_resolve_exception(es_clear, db, document_data_ref):
        """Test document person resolve."""
        with pytest.raises(Exception):
            Document.create(
                data=document_data_ref,
                delete_pid=False,
                dbcommit=True,
>               reindex=True
            )
E           Failed: DID NOT RAISE <class 'Exception'>

tests/ui/documents/test_documents_api.py:147: Failed
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -54,6 +54,7 @@ polib = "*"
 pycountry = ">=19.7.15"
 xmltodict = "*"
 redisbeat = "*"
+jsonpickle = "*"
 # TODO: to be removed when the thumbnail will be refactored
 angular-gettext-babel= ">=0.1"
 
@@ -94,7 +95,6 @@ docutils = "*"
 autoflake = ">=1.3.1"
 transifex-client = ">=0.12.5"
 appnope = { version = "*", optional = true }
-jsonpickle = "*"
 
 [tool.poetry.scripts]
 bootstrap = "scripts:run('./scripts/bootstrap')"

--- a/rero_ils/modules/documents/api.py
+++ b/rero_ils/modules/documents/api.py
@@ -182,12 +182,17 @@ class Document(IlsRecord):
 
     def index_persons(self, bulk=False):
         """Index all attached persons."""
+        from ..persons.api import Person
         persons_ids = []
         for author in self.get('authors', []):
+            person = None
             ref = author.get('$ref')
             if ref:
-                from ..persons.api import Person
                 person = Person.get_record_by_ref(ref)
+            pid = author.get('pid')
+            if pid:
+                person = Person.get_record_by_pid(pid)
+            if person:
                 if bulk:
                     persons_ids.append(person.id)
                 else:
@@ -213,9 +218,9 @@ class Document(IlsRecord):
         for idx, author in enumerate(authors):
             ref = author.get('$ref')
             if ref:
-                pers = Person.get_record_by_ref(ref)
-                if pers:
-                    authors[idx] = pers
+                person = Person.get_record_by_ref(ref)
+                if person:
+                    authors[idx] = person
         return super(Document, self).replace_refs()
 
 

--- a/scripts/__init__.py
+++ b/scripts/__init__.py
@@ -18,7 +18,7 @@
 """Poetry script utils."""
 
 import os
-import subprocess
+# import subprocess
 import sys
 
 # def __getattr__(name):  # python 3.7+, otherwise define each script manually

--- a/scripts/setup
+++ b/scripts/setup
@@ -92,7 +92,7 @@ ENABLE_WARNINGS=${ENABLE_WARNINGS:=true}
 # Displays program name
 msg "PROGRAM: ${PROGRAM}"
 
-# PIPENV is a mandatory condition to launch this program!
+# POETRY is a mandatory condition to launch this program!
 if [[ -z "${VIRTUAL_ENV}" ]]; then
   error_msg+exit "Error - Launch this script via poetry command:\n\tpoetry run ${PROGRAM}"
 fi
@@ -162,7 +162,7 @@ celery purge -A rero_ils.celery -f
 # Clean redis
 info_msg "Clean redis"
 eval "${PREFIX} invenio shell --no-term-title -c \"import redis; redis.StrictRedis.from_url(app.config['CACHE_REDIS_URL']).flushall(); print('Cache cleared')\""
-eval ${PREFIX} pipenv run invenio scheduler init -r
+eval ${PREFIX} invenio scheduler init -r
 
 eval ${PREFIX} invenio db destroy --yes-i-know
 eval ${PREFIX} invenio db init create
@@ -370,8 +370,8 @@ then
     eval ${PREFIX} invenio oaiharvester harvest -n ebooks -q -k
 else
     eval ${PREFIX} invenio scheduler enable_tasks -n bulk-indexer -n notification-creation -n accounts -n clear_and_renew_subscriptions -v
-    info_msg "Start OAI harvesting 100"
-    eval ${PREFIX} invenio oaiharvester harvest -n ebooks -a max=100 -q
+    info_msg "For ebooks harvesting run:"
+    msg "\tinvenio oaiharvester harvest -n ebooks -a max=100 -q"
 fi
 
 if ${ES_MAPPING}

--- a/tests/ui/documents/test_documents_api.py
+++ b/tests/ui/documents/test_documents_api.py
@@ -139,6 +139,10 @@ def test_document_can_delete_with_loans(
 
 def test_document_person_resolve_exception(es_clear, db, document_data_ref):
     """Test document person resolve."""
+    document_data_ref['authors'] = [{
+        '$ref': 'https://mef.rero.ch/api/rero/XXXXXX',
+        'type': 'person'
+    }],
     with pytest.raises(Exception):
         Document.create(
             data=document_data_ref,

